docs/IWYUStyle.md

```diff
@@ -0,0 +1,242 @@
+# IWYU Coding Style #
+
+IWYU is developed by a small number of loosely knit maintainers, and we rely on
+external contributions to stand a chance to improve the tool.
+
+We use code reviews as a stage gate for every patch, via the GitHub pull request
+system. We try our best to make reviews constructive and avoid power struggles,
+so please accept feedback in the good faith it was delivered.
+
+This document describes the most important of the conventions that contributors
+are expected to follow, so as to help make code reviews more productive and the
+code base more consistent and maintainable.
+
+
+## License headers ##
+
+Every source file must have a license header on the form:
+
+```
+//===--- <filename> - <description> ---------------------------------------===//
+//
+//                     The LLVM Compiler Infrastructure
+//
+// This file is distributed under the University of Illinois Open Source
+// License. See LICENSE.TXT for details.
+//
+//===----------------------------------------------------------------------===//
+```
+
+(With the corresponding comment prefix/suffix for other languages.)
+
+
+## C++ ##
+
+IWYU originated within Google, so its basic style is based on what Google had in
+their style guide at that time. Please refer to the [Google C++ style guide][1]
+for details, but note that newer decisions may not apply to IWYU. The most
+important points:
+
+* At most 80 characters per line
+* 2 spaces indentation, 4 spaces for continued lines
+* Unix line endings
+* PascalCase for types and functions, hacker_case for variables,
+  LOUDMOUTH_CASE for macros
+* Pointer asterisk and reference ampersand binds to the type name, not the
+  declared variable (i.e. declarations look like `Foo* f;` and `Bar& b;`)
+* When in doubt, mimic the code around you.
+
+IWYU has a `.clang-format` file, so everything should play out nicely if you use
+clang-format on your edited lines (but PLEASE do not reformat more than lines
+you touch, especially not whole files).
```

Comment:
On this topic you can mention the git-clang-format script from LLVM (https://github.com/llvm-mirror/clang/blob/master/tools/clang-format/git-clang-format) to help format commits and only changed lines.